lib/rspec/mocks/mutate_const.rb

```diff
@@ -105,6 +105,10 @@ class ConstantMutator
       #  so you can stub constants in other contexts (e.g. helper
       #  classes).
       def self.stub(constant_name, value, options={})
+        raise ArgumentError,
+              "Please provide a string as constant name" \
+              " You provided a #{constant_name.class.name}" unless constant_name.is_a?(String)
```

Comment:
I think it's confusing to put an `unless` after a multi-line expression -- instead, I think it would be more readable to put it first.

Also, we prefer doing type checks with `===`, e.g. `String === constant_name`.  The reason is that `constant_name` could be _anything_ and may not have a `is_a?` method or it may not work as we expect due to the user overriding it to behave differently -- but we expect `String.===` to always work correctly.  If you're monkeying with a core method on a core time we can't really support you.  A concrete example is a `BasicObject`. `BasicObject.new.is_a?(String)` fails with a `NoMethodError` but `String === BasicObject.new` correctly returns `false`.

Putting these together, this would be:

``` ruby
unless String === constant_name
  raise # ...
end
```